### PR TITLE
Use full version in actions to prevent auto-updates

### PR DESCRIPTION
**Pull request by mdegat01**
## Proposed change

Dependabot set a lot of our actions to auto-update within the latest major version. I looked up the latest version of each action we use and set the version to the full semantic version to prevent unexpected CI issues.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:
- Link to cli pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [ ] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Review by ludeeus — APPROVED** [review]

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
   stale:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/stale@v4
+      - uses: actions/stale@v4.1.0
```
Comment (mdegat01) [inline .github/workflows/stale.yml:12]: This actually isn't the latest version of `actions/stale`. The latest version is [v5.0.0](https://github.com/actions/stale/releases/tag/v5.0.0). I'm not really sure why dependabot hasn't tried to bump this one for us but I figured I'd leave it for a separate PR. The point of this PR is simply to correct the version numbers, not to change anything.
